src/components/CourseSideBar.vue

```diff
@@ -34,12 +38,25 @@ const navs = [
 </script>
 
 <template>
-  <ul class="menu menu-compact w-40 bg-base-100 p-2 lg:menu-normal">
-    <li
-      v-for="{ name, path } in navs"
-      :class="[{ 'border-l-4 border-blue-500': $route.path === `/course/${$route.params.name}${path}` }]"
-    >
-      <router-link :to="`/course/${$route.params.name}${path}`">{{ name }}</router-link>
-    </li>
-  </ul>
+  <template v-if="isDesktop">
+    <ul class="menu menu-compact w-40 bg-base-100 p-2 lg:menu-normal">
```

Comment:
the content in `<template v-if="isDesktop">` is single-root so we can eliminate the `<template />` use